src/main/scala/com/microsoft/partnercatalyst/fortis/spark/ProjectFortis.scala

```diff
@@ -0,0 +1,194 @@
+  }
+
+  private def createStreamingContext(): StreamingContext = {
+    val conf = new SparkConf()
+      .setAppName(Constants.SparkAppName)
+      .setIfMissing("spark.master", Constants.SparkMasterDefault)
+
+    implicit val streamingContext = new StreamingContext(
+      new SparkContext(conf),
+      Seconds(Constants.SparkStreamingBatchSizeDefault))
+
+    val streamProvider = StreamProviderFactory.create()
+    val streamRegistry = getStreamRegistry
+
+    // Attach each pipeline (aka code path)
+    // 'fortisEvents' is the stream of analyzed data aggregated (union) from all pipelines
+    val fortisEvents = pipelines.flatMap(
+      pipeline => pipeline(streamProvider, streamRegistry)
+    ).reduceOption(_.union(_))
+
+    // TODO: other computations and save to DB
+    fortisEvents.foreach(
+      _.foreachRDD(
+        _.foreachPartition(throw new NotImplementedException)))
+
+    streamingContext
+  }
+
+  // Main starts here
+  val ssc = StreamingContext.getOrCreate(Settings.progressDir, createStreamingContext)
+
+  ssc.start()
+  ssc.awaitTermination()
+
+  /**
+    * TODO: Use Cassandra to populate
+    * Build connector config registry from hard-coded values for demo.
+    *
+    * The key is the name of the pipeline and the value is a list of connector configs whose streams should comprise it.
+    */
+  private def getStreamRegistry: Map[String, List[ConnectorConfig]] = {
+    Map[String, List[ConnectorConfig]](
+      "instagram" -> List(
+        ConnectorConfig(
+          "InstagramTag",
+          Map(
+            "authToken" -> System.getenv("INSTAGRAM_AUTH_TOKEN"),
+            "tag" -> "rose"
+          )
+        ),
+        ConnectorConfig(
+          "InstagramLocation",
+          Map(
+            "authToken" -> System.getenv("INSTAGRAM_AUTH_TOKEN"),
+            "latitude" -> "49.25",
+            "longitude" -> "-123.1"
+          )
+        )
+      ),
+      "radio" -> List(
+        ConnectorConfig(
+          "Radio",
+          Map(
+            "subscriptionKey" -> System.getenv("OXFORD_SPEECH_TOKEN"),
+            "radioUrl" -> "http://live02.rfi.fr/rfimonde-32.mp3", // "http://bbcmedia.ic.llnwd.net/stream/bbcmedia_radio3_mf_p",
+            "audioType" -> ".mp3",
+            "locale" -> "fr-FR", //"en-US",
+            "speechType" -> "CONVERSATION",
+            "outputFormat" -> "SIMPLE"
+          )
+        )
+      ),
+      "twitter" -> List(
+        ConnectorConfig(
+          "Twitter",
+          Map(
+            "consumerKey" -> System.getenv("TWITTER_CONSUMER_KEY"),
+            "consumerSecret" -> System.getenv("TWITTER_CONSUMER_SECRET"),
+            "accessToken" -> System.getenv("TWITTER_ACCESS_TOKEN"),
+            "accessTokenSecret" -> System.getenv("TWITTER_ACCESS_TOKEN_SECRET")
+          )
+        )
+      ),
+      "facebook" -> List(
+        ConnectorConfig(
+          "FacebookPage",
+          Map(
+            "accessToken" -> System.getenv("FACEBOOK_AUTH_TOKEN"),
```

Comment:
Might as well use the getenvOrFail helper created previously since these environment variables are all required if the connector is used. May need changing the map from `Map[String,String]` to `Map[String,()=>String]` so that the values are lazy-loaded.

On second thoughts: shouldn't these all be part of the settings since they'll also have to come from Cassandra going forward?